### PR TITLE
[WIP] Target net core 2

### PR DESCRIPTION
**Pull request by ChrisJollyAU**
Change target framework to run fully on .Net Core 2 and Entity Framework Core 2

Major changes include:
- Target Net Core 2
- Change to Entity Framework Core 2
- Remove SQL Compact Edition
- Add MySQL, SQLite and PostgreSQL providers
 - Change to SixLabors.ImageSharp for image processing
 - Minor edits to make sure it runs on unix systems

For more info see https://www.nopcommerce.com/boards/t/48993/targetting-net-core-2-aka-net-core-2-running-on-net-core-2.aspx

### CODE REVIEW

File: src/Libraries/Nop.Core/Domain/Catalog/Product_ProductTag_Mappng.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Nop.Core.Domain.Catalog
+{
+    public class Product_ProductTag_Mappng
```
Comment (dahlbyk) [inline src/Libraries/Nop.Core/Domain/Catalog/Product_ProductTag_Mappng.cs]: Mappng :point_right: Mapping?

Comment (ChrisJollyAU) [reply src/Libraries/Nop.Core/Domain/Catalog/Product_ProductTag_Mappng.cs]: Will fix that spelling sometime in the next couple of days

Comment (ChrisJollyAU) [reply src/Libraries/Nop.Core/Domain/Catalog/Product_ProductTag_Mappng.cs]: committed fix over the weekend

**Comment by AndreiMaz** [comment]
We've already moved .NET Core. But we leave this pull reqeust open because it could be helpful for us in the future (when we start support new databases)

**Pull request closed without merging**